[gradle] BasicJavaNativeBuildIT fails with latest JDK 17-based graal master

### Describe the bug

Something seems wrong with the project it fails fairly early on with:

```
[INFO] Running io.quarkus.gradle.nativeimage.BasicJavaNativeBuildIT
$ /home/runner/work/mandrel/mandrel/quarkus/integration-tests/gradle/./gradlew -Dmaven.repo.local=/home/runner/.m2/repository -Dorg.gradle.console=plain -Dorg.gradle.daemon=false --configuration-cache --stacktrace --info --daemon clean buildNative -Dquarkus.package.type=fast-jar
Command: /home/runner/work/mandrel/mandrel/quarkus/integration-tests/gradle/./gradlew -Dmaven.repo.local=/home/runner/.m2/repository -Dorg.gradle.console=plain -Dorg.gradle.daemon=false --configuration-cache --stacktrace --info --daemon clean buildNative -Dquarkus.package.type=fast-jar ran from: /home/runner/work/mandrel/mandrel/quarkus/integration-tests/gradle/target/classes/basic-java-native-module failed with exit code: 1 and the following output:

FAILURE: Build failed with an exception.

* What went wrong:
Configuration cache problems found in this build.

1 problem was found storing the configuration cache.
- Task `:buildNative` of type `org.gradle.api.DefaultTask`: cannot serialize object of type 'org.gradle.api.internal.project.DefaultProject', a subtype of 'org.gradle.api.Project', as these are not supported with the configuration cache.
  See https://docs.gradle.org/8.0.2/userguide/configuration_cache.html#config_cache:requirements:disallowed_types

See the complete report at file:///home/runner/work/mandrel/mandrel/quarkus/integration-tests/gradle/target/classes/basic-java-native-module/build/reports/configuration-cache/w5yvnic768h3j466x2dt69mu/6dwwq3v53qeev1ux14bzo0nae/configuration-cache-report.html

* Try:
> Run with --debug option to get more log output.
> Run with --scan to get full insights.

* Exception is:
org.gradle.configurationcache.ConfigurationCacheProblemsException: Configuration cache problems found in this build.

1 problem was found storing the configuration cache.
- Task `:buildNative` of type `org.gradle.api.DefaultTask`: cannot serialize object of type 'org.gradle.api.internal.project.DefaultProject', a subtype of 'org.gradle.api.Project', as these are not supported with the configuration cache.
  See https://docs.gradle.org/8.0.2/userguide/configuration_cache.html#config_cache:requirements:disallowed_types

See the complete report at file:///home/runner/work/mandrel/mandrel/quarkus/integration-tests/gradle/target/classes/basic-java-native-module/build/reports/configuration-cache/w5yvnic768h3j466x2dt69mu/6dwwq3v53qeev1ux14bzo0nae/configuration-cache-report.html
	at org.gradle.configurationcache.problems.ConfigurationCacheProblems.report(ConfigurationCacheProblems.kt:161)
	at org.gradle.internal.buildtree.ProblemReportingBuildActionRunner.reportProblems(ProblemReportingBuildActionRunner.java:61)
	at org.gradle.internal.buildtree.ProblemReportingBuildActionRunner.run(ProblemReportingBuildActionRunner.java:52)
	at org.gradle.launcher.exec.BuildOutcomeReportingBuildActionRunner.run(BuildOutcomeReportingBuildActionRunner.java:65)
	at org.gradle.tooling.internal.provider.FileSystemWatchingBuildActionRunner.run(FileSystemWatchingBuildActionRunner.java:140)
	at org.gradle.launcher.exec.BuildCompletionNotifyingBuildActionRunner.run(BuildCompletionNotifyingBuildActionRunner.java:41)
	at org.gradle.launcher.exec.RootBuildLifecycleBuildActionExecutor.lambda$execute$0(RootBuildLifecycleBuildActionExecutor.java:40)
	at org.gradle.composite.internal.DefaultRootBuildState.run(DefaultRootBuildState.java:122)
	at org.gradle.launcher.exec.RootBuildLifecycleBuildActionExecutor.execute(RootBuildLifecycleBuildActionExecutor.java:40)
	at org.gradle.internal.buildtree.DefaultBuildTreeContext.execute(DefaultBuildTreeContext.java:40)
	at org.gradle.launcher.exec.BuildTreeLifecycleBuildActionExecutor.lambda$execute$0(BuildTreeLifecycleBuildActionExecutor.java:65)
	at org.gradle.internal.buildtree.BuildTreeState.run(BuildTreeState.java:53)
	at org.gradle.launcher.exec.BuildTreeLifecycleBuildActionExecutor.execute(BuildTreeLifecycleBuildActionExecutor.java:65)
	at org.gradle.launcher.exec.RunAsBuildOperationBuildActionExecutor$3.call(RunAsBuildOperationBuildActionExecutor.java:61)
	at org.gradle.launcher.exec.RunAsBuildOperationBuildActionExecutor$3.call(RunAsBuildOperationBuildActionExecutor.java:57)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
	at org.gradle.launcher.exec.RunAsBuildOperationBuildActionExecutor.execute(RunAsBuildOperationBuildActionExecutor.java:57)
	at org.gradle.launcher.exec.RunAsWorkerThreadBuildActionExecutor.lambda$execute$0(RunAsWorkerThreadBuildActionExecutor.java:36)
	at org.gradle.internal.work.DefaultWorkerLeaseService.withLocks(DefaultWorkerLeaseService.java:249)
	at org.gradle.internal.work.DefaultWorkerLeaseService.runAsWorkerThread(DefaultWorkerLeaseService.java:109)
	at org.gradle.launcher.exec.RunAsWorkerThreadBuildActionExecutor.execute(RunAsWorkerThreadBuildActionExecutor.java:36)
	at org.gradle.tooling.internal.provider.continuous.ContinuousBuildActionExecutor.execute(ContinuousBuildActionExecutor.java:110)
	at org.gradle.tooling.internal.provider.SubscribableBuildActionExecutor.execute(SubscribableBuildActionExecutor.java:64)
	at org.gradle.internal.session.DefaultBuildSessionContext.execute(DefaultBuildSessionContext.java:46)
	at org.gradle.tooling.internal.provider.BuildSessionLifecycleBuildActionExecuter$ActionImpl.apply(BuildSessionLifecycleBuildActionExecuter.java:100)
	at org.gradle.tooling.internal.provider.BuildSessionLifecycleBuildActionExecuter$ActionImpl.apply(BuildSessionLifecycleBuildActionExecuter.java:88)
	at org.gradle.internal.session.BuildSessionState.run(BuildSessionState.java:69)
	at org.gradle.tooling.internal.provider.BuildSessionLifecycleBuildActionExecuter.execute(BuildSessionLifecycleBuildActionExecuter.java:62)
	at org.gradle.tooling.internal.provider.BuildSessionLifecycleBuildActionExecuter.execute(BuildSessionLifecycleBuildActionExecuter.java:41)
	at org.gradle.tooling.internal.provider.StartParamsValidatingActionExecuter.execute(StartParamsValidatingActionExecuter.java:63)
	at org.gradle.tooling.internal.provider.StartParamsValidatingActionExecuter.execute(StartParamsValidatingActionExecuter.java:31)
	at org.gradle.tooling.internal.provider.SessionFailureReportingActionExecuter.execute(SessionFailureReportingActionExecuter.java:50)
	at org.gradle.tooling.internal.provider.SessionFailureReportingActionExecuter.execute(SessionFailureReportingActionExecuter.java:38)
	at org.gradle.tooling.internal.provider.SetupLoggingActionExecuter.execute(SetupLoggingActionExecuter.java:47)
	at org.gradle.tooling.internal.provider.SetupLoggingActionExecuter.execute(SetupLoggingActionExecuter.java:31)
	at org.gradle.launcher.daemon.server.exec.ExecuteBuild.doBuild(ExecuteBuild.java:65)
	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:37)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
	at org.gradle.launcher.daemon.server.exec.WatchForDisconnection.execute(WatchForDisconnection.java:39)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
	at org.gradle.launcher.daemon.server.exec.ResetDeprecationLogger.execute(ResetDeprecationLogger.java:29)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
	at org.gradle.launcher.daemon.server.exec.RequestStopIfSingleUsedDaemon.execute(RequestStopIfSingleUsedDaemon.java:35)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
	at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.create(ForwardClientInput.java:78)
	at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.create(ForwardClientInput.java:75)
	at org.gradle.util.internal.Swapper.swap(Swapper.java:38)
	at org.gradle.launcher.daemon.server.exec.ForwardClientInput.execute(ForwardClientInput.java:75)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
	at org.gradle.launcher.daemon.server.exec.LogAndCheckHealth.execute(LogAndCheckHealth.java:64)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
	at org.gradle.launcher.daemon.server.exec.LogToClient.doBuild(LogToClient.java:63)
	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:37)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
	at org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment.doBuild(EstablishBuildEnvironment.java:84)
	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:37)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
	at org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy$1.run(StartBuildOrRespondWithBusy.java:52)
	at org.gradle.launcher.daemon.server.DaemonStateCoordinator$1.run(DaemonStateCoordinator.java:297)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:49)


* Get more help at https://help.gradle.org/

BUILD FAILED in 4m 57s
```

See:
https://github.com/graalvm/mandrel/actions/runs/4705326599/jobs/8345996009#step:11:171

### Expected behavior

Build works.

### Actual behavior

Build fails with:

```
cannot serialize object of type 'org.gradle.api.internal.project.DefaultProject', a subtype of 'org.gradle.api.Project', as these are not supported with the configuration cache.
```

### How to Reproduce?

Run the native gradle integration test with a recent Mandrel build from GraalVM master. E.g. from here:
https://github.com/graalvm/mandrel/suites/12252475032/artifacts/648689931


### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

JDK 17.0.7-ea

### GraalVM version (if different from Java)

Master

### Quarkus version or git rev

main

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/integration-tests/gradle/src/test/java/io/quarkus/gradle/nativeimage/BasicJavaNativeBuildIT.java
+++ b/integration-tests/gradle/src/test/java/io/quarkus/gradle/nativeimage/BasicJavaNativeBuildIT.java
@@ -24,13 +24,8 @@ public class BasicJavaNativeBuildIT extends QuarkusNativeGradleITBase {
         assertThat(build.getTasks().get(":quarkusBuild")).isEqualTo(BuildResult.SUCCESS_OUTCOME);
         final String buildOutput = build.getOutput();
         // make sure the output log during the build contains some expected logs from the native-image process
-        CharSequence[] expectedOutput;
-        if (buildOutput.contains("Version info:")) { // Starting with 22.0 the native-image output changed
-            expectedOutput = new CharSequence[] { "Initializing...", "Performing analysis...",
-                    "Finished generating '" + NATIVE_IMAGE_NAME + "' in" };
-        } else {
-            expectedOutput = new CharSequence[] { "(clinit):", "(typeflow):", "[total]:" };
-        }
+        CharSequence[] expectedOutput = new CharSequence[] { "Initializing...", "Performing analysis...",
+                "Finished generating '" + NATIVE_IMAGE_NAME + "' in" };
         assertThat(buildOutput)
                 .withFailMessage("native-image build log is missing certain expected log messages: \n\n %s", buildOutput)
                 .contains(expectedOutput);
@@ -58,13 +53,8 @@ public class BasicJavaNativeBuildIT extends QuarkusNativeGradleITBase {
         assertThat(build.getTasks().get(":quarkusBuild")).isEqualTo(BuildResult.SUCCESS_OUTCOME);
         final String buildOutput = build.getOutput();
         // make sure the output log during the build contains some expected logs from the native-image process
-        CharSequence[] expectedOutput;
-        if (buildOutput.contains("Version info:")) { // Starting with 22.0 the native-image output changed
-            expectedOutput = new CharSequence[] { "Initializing...", "Performing analysis...",
-                    "Finished generating 'test-runner' in" };
-        } else {
-            expectedOutput = new CharSequence[] { "(clinit):", "(typeflow):", "[total]:" };
-        }
+        CharSequence[] expectedOutput = new CharSequence[] { "Initializing...", "Performing analysis...",
+                "Finished generating 'test-runner' in" };
         assertThat(buildOutput)
                 .withFailMessage("native-image build log is missing certain expected log messages: \n\n %s", buildOutput)
                 .contains(expectedOutput)
@@ -93,13 +83,8 @@ public class BasicJavaNativeBuildIT extends QuarkusNativeGradleITBase {
         assertThat(build.getTasks().get(":quarkusBuild")).isEqualTo(BuildResult.SUCCESS_OUTCOME);
         final String buildOutput = build.getOutput();
         // make sure the output log during the build contains some expected logs from the native-image process
-        CharSequence[] expectedOutput;
-        if (buildOutput.contains("Version info:")) { // Starting with 22.0 the native-image output changed
-            expectedOutput = new CharSequence[] { "Initializing...", "Performing analysis...",
-                    "Finished generating 'test' in" };
-        } else {
-            expectedOutput = new CharSequence[] { "(clinit):", "(typeflow):", "[total]:" };
-        }
+        CharSequence[] expectedOutput = new CharSequence[] { "Initializing...", "Performing analysis...",
+                "Finished generating 'test' in" };
         assertThat(buildOutput)
                 .withFailMessage("native-image build log is missing certain expected log messages: \n\n %s", buildOutput)
                 .contains(expectedOutput)

